src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -125,6 +125,12 @@ trait Types
   def skolemizationLevel = _skolemizationLevel
   def skolemizationLevel_=(value: Int) = _skolemizationLevel = value
 
+  /** The current level of active TypeVars. Any TypeVar created at an earlier level will not accumulate constraints.
+    */
+  private[this] var _typeVarSuspensionLevel = 0
+  def typeVarSuspensionLevel: Int = _typeVarSuspensionLevel
+  def typeVarSuspensionLevel_=(value: Int) = _typeVarSuspensionLevel = value
```

Comment:
This will need an override in `SynchronizedTypes` to re-implement in a thread-local manner.